test/dashboard-e2e/api/api-helpers.js

```diff
@@ -1,17 +1,26 @@
 import superagent from 'superagent';
 
 export class ApiHelpers {
-    constructor(apiUrl) {
+    constructor(apiUrl, cloudContext='', bearerToken='') {
         this.API_URL = apiUrl;
+        this.CLOUD_CONTEXT = cloudContext;
+        this.BEARER_TOKEN = bearerToken;
     }
 
     async getTests() {
         const request = `${this.API_URL}/tests`;
 
         try {
-            const response = await superagent.get(request);
+            if(!this.CLOUD_CONTEXT) {
+                const response = await superagent.get(request);
 
-            return response.body;
+                return response.body;
+            } 
+                const response = await superagent.get(request)
+                    .set('Authorization', `Bearer ${this.BEARER_TOKEN}`);
+
+                return JSON.parse(response.text);
```

Comment:
* You may either use `response.body` or `JSON.parse(response.text)`, but both should be equivalent for JSON responses, so no need to use both
* `CLOUD_CONTEXT` is not needed - it could depend on the `BEARER_TOKEN`

```ts
  const response = await superagent.get(request)
    .set('authorization', this.BEARER_TOKEN ? `Bearer ${this.BEARER_TOKEN}` : undefined);
  return JSON.parse(response.text);
```

Same for all other functions

Reply:
Additionally, this part could be split to separate function, but it may be done later - we should anyway change the `api-helpers` to TypeScript file.

```ts
async call(method, request) {
  const response = await superagent[method.toLowerCase()](request)
    .set('content-type', 'application/json')
    .set('authorization', this.BEARER_TOKEN ? `Bearer ${this.BEARER_TOKEN}` : undefined);
  return JSON.parse(response.text);
}

// and usage:
await this.call('GET', `${this.API_URL}/tests`)
```

Reply:
I used `JSON.parse(response.text)` in case of Cloud instead of just `response.body` because despite setting content type the empty body is returned.
```
getTests response.body: 
{}
getTests response.text: 
[{"name":"artillery-executor-smoke","namespace":"testkube-dev","type":"artillery/test","content":{"type":"git","repository":{"type":"git","uri":"https://github.com/kubeshop/testkube.git","branch":"main","path":"test/artillery/executor-smoke/artillery-smoke-test.yaml"}},"created":"2023-05-23T00:03:08Z","labels":{"core-tests":"executors","executor":"artillery-executor","test-ty
...
```

Reply:
I merged Authentication part, and moved response parsing to separate `parseResponse` function.